assets/app/scripts/directives/logViewer.js

```diff
@@ -64,6 +64,32 @@ angular.module('openshiftConsole')
               logLinks.scrollTop();
             };
 
+            var bufferedLines = [];
+            // Display all buffered lines.
+            var update = function() {
+              var elements = [];
+              angular.forEach(bufferedLines, function(line) {
+                // Manipulate the DOM directly for better performance displaying large log files.
+                var logLineDiv = $('<div row class="log-line"/>');
```

Comment:
You could even make `bufferedLines` the document fragment and append to it, then throttle the append of the buffer fragment to the actual body DOM node, to avoid an intermediate buffer